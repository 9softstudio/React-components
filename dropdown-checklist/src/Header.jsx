import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Header extends Component {
    onClickHandler = () => {
        this.props.onClickHandler();
    }
    render() {
        const { dropdownName, opened, selectedTextElement, headerRef} = this.props;
        return (
            <div ref={headerRef} 
                id={dropdownName}
                className={"laForm__field fdcl " + (opened ? "opened" : "")}
                onClick={this.onClickHandler}>
                <span className="fdcl__text">{selectedTextElement}</span>
            </div>
        );
    }
};