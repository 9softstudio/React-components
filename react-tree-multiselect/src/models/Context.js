import React from 'react';

export default React.createContext({
    rootNode: {},
    nodes: [],
    onToggle: () => { },
    onCheck: () => { }
});