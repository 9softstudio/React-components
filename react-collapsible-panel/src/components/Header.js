import React from 'react';
import { mergeClassName } from '../utils';
import Context from '../panel-context';

export default (props) => {
    return (<Context.Consumer>{({ onToggleCollapse, collapsible }) => (
        <div className={mergeClassName(props, 'la-panel__header')}>
            {collapsible && (<span className="la-panel__indicator" onClick={onToggleCollapse} />)}
            {props.children}
        </div>
    )}</Context.Consumer>);
};