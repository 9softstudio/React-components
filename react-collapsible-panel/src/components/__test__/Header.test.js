import React from 'react';
import Header from '../Header';
import Context from '../../panel-context';
import renderer from 'react-test-renderer';

describe('Panel Header is collapsible:', () => {
    test('renders correctly', () => {
        const component = renderer
            .create(
                <Context.Provider value={({ collapsible: true })}>
                    <Header />
                </Context.Provider>
            );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('custom class name', () => {
        const component = renderer
            .create(
                <Context.Provider value={({ collapsible: true })}>
                    <Header className="hello" />
                </Context.Provider>
            );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('has children', () => {
        const component = renderer
            .create(
                <Context.Provider value={({ collapsible: true })}>
                    <Header className="hello">
                        <span>world</span>
                    </Header>
                </Context.Provider>
            );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('click on indicator: onToggleCollapse()', () => {
        const onToggleCollapse = jest.fn();
        const component = renderer
            .create(
                <Context.Provider value={({ collapsible: true, onToggleCollapse: onToggleCollapse })}>
                    <Header className="hello">
                        <span>world</span>
                    </Header>
                </Context.Provider>
            );
        expect(component.toJSON().children[0].props.onClick).toBe(onToggleCollapse);
    });
});

test('Panel Header is not collapsible: hide indicator icon', () => {
    const component = renderer
        .create(
            <Context.Provider value={({ collapsible: false })}>
                <Header className="hello">
                    <span>world</span>
                </Header>
            </Context.Provider>
        );
    expect(component.toJSON()).toMatchSnapshot();
});