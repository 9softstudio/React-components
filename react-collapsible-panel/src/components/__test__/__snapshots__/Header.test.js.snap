// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel Header is collapsible: custom class name 1`] = `
<div
  className="hello la-panel__header"
>
  <span
    className="la-panel__indicator"
  />
</div>
`;

exports[`Panel Header is collapsible: has children 1`] = `
<div
  className="hello la-panel__header"
>
  <span
    className="la-panel__indicator"
  />
  <span>
    world
  </span>
</div>
`;

exports[`Panel Header is collapsible: renders correctly 1`] = `
<div
  className="la-panel__header"
>
  <span
    className="la-panel__indicator"
  />
</div>
`;

exports[`Panel Header is not collapsible: hide indicator icon 1`] = `
<div
  className="hello la-panel__header"
>
  <span>
    world
  </span>
</div>
`;
